applications/Preprocessor/lib/output_impl.h

```diff
@@ -102,11 +103,18 @@ void outputMesh(Mesh<dimension>& mesh,
     }
     std::ofstream output(outputFileName.getValue());
     output << std::hexfloat;
-    output << "mesh 1" << std::endl;
+    size_t version = 2;
+    output << "mesh " << version << std::endl;
     output << mesh.getNumberOfNodes() << " " << mesh.getNumberOfElements()
            << " " << dimension;
     printOtherEntityCounts(output, mesh, Detail::tag<dimension - 1>{});
     output << std::endl;
+    output << "zones" << std::endl;
+    for (const auto& zonename : mesh.getZoneNames()) {
+        std::cout << zonename << '\n';
+    }
+
+    output << "nodes" << std::endl;
     std::size_t reservedSpace =
```

Comment:
The ordering of the output is incorrect here. It should have been:
 - Partition information (this is old lines 110-119)
 - Zone information